src/certgen_accessor.cc

```diff
@@ -19,16 +19,91 @@
     =========================================================================
 */
 
-/*
-@header
-    certgen_accessor -
-@discuss
-@end
-*/
+#include "fty_certificate_generator_classes.h"
+#include "certgen_accessor.h"
 
 //  Structure of our class
 namespace certgen
 {
+    CertGenAccessor::CertGenAccessor(fty::SyncClient & reqClient)
+        : m_requestClient (reqClient)
+    {}
+
+    void CertGenAccessor::generateSelfCertificateReq(
+        const std::string & serviceName
+    ) const
+    {
+        fty::Payload payload;
+        try
+        {
+            payload = sendCommand(GENERATE_SELFSIGNED_CERTIFICATE, {serviceName});
+        }
+        catch (const std::exception& e)
+        {
+            log_info("certificate generator accessor returned '%s'", e.what());
+        }
+    }
+
+    std::string CertGenAccessor::generateCsr(const std::string & serviceName) const
+    {
+        fty::Payload payload;
+        try
+        {
+            payload = sendCommand(GENERATE_CSR, {serviceName});
+        }
+        catch (const std::exception& e)
+        {
+            log_info("certificate generator accessor returned '%s'", e.what());
```

Comment:
Do not log anything at library level. The user of the library should decide what to do with the exception